### ISSUE TITLE
Heading color only targets .elementor-heading-title 

### ISSUE THREAD

**Issue by kokers** [pojome/elementor#2866]
When changing colors inside Style > Typography, headings should change colors for all h and not only .elementor-heading-title. http://prntscr.com/hh7c8b

If someone uses multiple text editors, like for example 3 columns with image box widget (http://prntscr.com/hh7dc7) or simple text editor with content and headings inside - then changing heading color should also be changed for them. It's not that uncommon to have different color for headings.

Bottom line, changing heading color should target .elementor-heading-title and h1,h2,h3,h4,h5 inside that section/column.

Currently I need to add custom css like:

`selector h1, selector h2, selector h3{color:white};`

in order to apply that color.

**Comment by bainternet** [comment]
This is not the way Elementor works (for now).

Each widget style is configured on its own and we do not overwrite theme's stylesheet defaults.

I do understand the use-case your bring where it would actually save you time but it's not up to a single widget to overwrite the whole page.

Thanks for your suggestion.

**Comment by kokers** [comment]
@bainternet Then what's the point on having option "Heading Color" inside Section > Style > Typography? 
All other options "Text Color", "Link Color", "Link Hover Color" works (almost) as expected = they are changing colors for all widgets inside that section. 

Please look at this screenshot http://prntscr.com/hh7ocw I have 4 different widgets inside section, and I've changed colors for text and link and the only that is not working is Heading Color. And I as a user expect that headings (h1,h2,h3,h4) should be changed to that color. Unless it's overwrite inside widget itself later on. So basically the hierarchy would be theme colors > section colors > widget colors. 

I honestly don't know what't the point of having "Heading Color" option inside Style that simply doesn't work. And if it's not targeted to headings, then it should have different name as it's really confusing.

**Comment by kokers** [comment]
@bainternet can you please follow up on this? I'm really interested what logic is behind this, as targeting only ".elementor-heading-title" is in contradiction with "Each widget style is configured on its own and we do not overwrite theme's stylesheet defaults.".

**Comment by bainternet** [comment]
@kokers 

Got redirected here from the support, let's try to make sense of how things work:

When you set a color in "Edit section" it sets a default color for all **Heading widgets** in that section regardless of what HTML tag is used ( H1, H2.... ) as long as it is a Heading widget:

![jrxlyso](https://user-images.githubusercontent.com/660572/33655261-42fe0d54-da7b-11e7-8e85-f01d23deb219.png)

When you set a color in each widget style you only change it for that widget:

![zomotvd](https://user-images.githubusercontent.com/660572/33655372-86582058-da7b-11e7-93db-f13cae56adfd.png)


This way you can set a color "Globally" per that section, and in each widget, you can customize the color of that specific widget.

So basically when you set "Heading Color" in a section, It is only intended for **Heading widgets** inside that section and does not overwrite the theme styles, simply sets the style of the heading widgets in that section.

And I agree with you about the naming that could be changed to avoid confusions.

Hope this has cleared a bit up.

**Comment by kokers** [comment]
@bainternet The problem I have with that is that I don't understand why in that settings you targeting ONLY headings inside heading widget (that ONE particular widget) and not other headings inside other widgets (just like you are doing this with Text color). 

When you are setting Text color you are in fact overwriting themes text color but for this particular section. I honestly can't see any reason why not to allow the same wit heading color to overwrite heading colors inside that section (exactly like it's done for texts AND links). This is extremely confusing as I am developer myself and see whats behind the scenes. 

When you set text color the code generated looks similar to this:
```
.elementor-2622 .elementor-element.elementor-element-14dddd3 {
	 color: #23a455;
}
```

For links:
```
.elementor-2622 .elementor-element.elementor-element-14dddd3 a {
	color: #4054b2;
}
```

So in fact, you don't change theme style, but section style. With targeted ID. 

Yet for headings it's only targeting widget and not typography styles:
```
.elementor-2622 .elementor-element.elementor-element-14dddd3 .elementor-heading-title {
	color: #6ec1e4;
}
```

I don't understand why you can't make this heading setting work like this:
```
.elementor-2622 .elementor-element.elementor-element-14dddd3 .elementor-heading-title,
.elementor-2622 .elementor-element.elementor-element-14dddd3 h1,
.elementor-2622 .elementor-element.elementor-element-14dddd3 h2,
.elementor-2622 .elementor-element.elementor-element-14dddd3 h3,
.elementor-2622 .elementor-element.elementor-element-14dddd3 h4,
.elementor-2622 .elementor-element.elementor-element-14dddd3 h5{
	color: #6ec1e4;
}
```

Which makes more sense, and you with that you truly change typography settings for that section instead of simply targeting one specific widget, which as you said "Each widget style is configured on its own and we do not overwrite theme's stylesheet defaults.". And currently there are mixed options. 3 settings from typography allow to change global typography settings for this one particular section, and one is targeting only a widget. 

And this confusion was already pointed out here https://github.com/pojome/elementor/issues/556 but also was very quickly closed.

Users not always use Heading widget for headings. They also use proper html tag inside WYSIWYG widget. And with current setup, you change color for text inside section settings which changes text color for all texts, but then you need to click through all widgets to change color for headings ONLY inside editors, because it's the only thing that you can't change for section even tho there is "headings color" option. Thats insane. 

If I'm alone with that view (I doubt it) fine, I will teach my clients that they need to write in Custom CSS code like from PRO:
```
selector h1,
selector h2,
selector h3,
selector h4,
selector h5{
      color:red;
}
```
and elementor will translate that later to `.elementor-2622  h1, .elementor-2622 h2, and so on` but it's not user friendly and doesn't make sense. But if there is more users that agree with that (and there is at least one from https://github.com/pojome/elementor/issues/556) then please consider making the adjustments. 

Ask your community.

**Comment by bainternet** [comment]
@kokers 

Like I said, I agree with you about the naming, if it was named "Heading widget color" instead of "heading color" then you would not have been confused or expecting different results then what you actually get. 
That control was and is set to define only the style of heading widgets in a section.

As for a new control for setting the H1-6 colors is currently not available, and we will look into it in the future if we see a large (enough) demand for it.

**Comment by kokers** [comment]
If anyone need this then here is a code that will add such control https://gist.github.com/kokers/3f00c72752e52cc5d6249ebe715b3008

**Comment by bainternet** [comment]
Glad you figured it out and thanks.

**Comment by yankiara** [comment]
Hi all,
Thanks a zillion @kokers, you made my day!!!
The current Elementor option is soooo limited ;-)
